### PR TITLE
TensorFlow 2.0 tf.estimator and tf.keras autologging support

### PR DESCRIPTION
**Pull request by juntai-zheng**
## What changes are proposed in this pull request?
 
In this PR, autologging for `tf.estimator` and `tf.keras` models has been updated to be compatible with TensorFlow 2.0. The same functionalities of autologging in TF 1.x will be offered in TF 2.0, namely when fitting `tf.keras` models and when exporting saved `tf.estimator` models. This is done via a new Callback class `__MLflowTfKeras2Callback()` which no longer uses TensorBoard for metrics (instead using `on_epoch_end()`) and uses `model.get_config()` for parameter fetching instead of manual searching.

In addition, a TF 1.X autologging test name was fixed, and travis is updated to use TF==1.14 instead of 1.12 for testing TF 1.X (the renamed test breaks with 1.12).
 
## How is this patch tested?
 
A new testing file, `test_tensorflow2_autolog.py` (based on `test_tensorflow_autolog.py`) was created and tests the new changes.
 
### Is this a user-facing change? 

- [ ] No. You can skip the rest of this section.
- [X] Yes. Give a description of this change to be included in the release notes for MLflow users.
 
MLflow now supports autologging for `tf.keras` and `tf.estimator` models in TensorFlow 2.0. You can see the documentation for autlogging [here](https://www.mlflow.org/docs/latest/tracking.html#automatic-logging-from-tensorflow-and-keras-experimental).
 
### What component(s) does this PR affect?
 
- [ ] UI
- [ ] CLI 
- [X] API 
- [ ] REST-API 
- [ ] Examples 
- [ ] Docs
- [X] Tracking
- [ ] Projects 
- [ ] Artifacts 
- [X] Models 
- [ ] Scoring 
- [ ] Serving
- [ ] R
- [ ] Java
- [X] Python

### How should the PR be classified in the release notes? Choose one:
 
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [X] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: tests/tensorflow_autolog/test_tensorflow2_autolog.py [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+
+    return client.get_run(run.info.run_id)
+
+
+@pytest.mark.large
+def test_tf_keras_autolog_logs_expected_data(tf_keras_random_data_run):
+    data = tf_keras_random_data_run.data
+    assert 'accuracy' in data.metrics
+    assert 'loss' in data.metrics
+    assert 'name' in data.params
```
Comment (smurching) [inline tests/tensorflow_autolog/test_tensorflow2_autolog.py]: Hm having a param called `'name'` is a bit confusing out of context, maybe we could preface each of the optimizer param names with `optimizer_`? So we'd have `optimizer_name`, `optimizer_learning_rate`, ... etc. If that results in the param name being truncated in the UI (would be good to try it out & confirm), maybe we could do `opt_name`, `opt_learning_rate`, etc instead.

File: tests/tensorflow_autolog/test_tensorflow2_autolog.py [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+    assert data.params['name'] == 'Adam'
+    assert 'learning_rate' in data.params
+    assert 'decay' in data.params
+    assert 'beta_1' in data.params
+    assert 'beta_2' in data.params
+    assert 'epsilon' in data.params
+    assert 'amsgrad' in data.params
+    assert data.params['amsgrad'] == 'False'
+    print(data.params)
```
Comment (smurching) [inline tests/tensorflow_autolog/test_tensorflow2_autolog.py]:
```suggestion
```

File: travis/run-large-python-tests.sh [hunk 3]
```diff
@@ -34,11 +34,12 @@ pytest --verbose tests/sagemaker/mock --large
 pytest --verbose tests/sklearn --large
 pytest --verbose tests/spark --large
 pytest --verbose tests/tensorflow/test_tensorflow_model_export.py --large
+pytest --verbose tests/tensorflow_autolog/test_tensorflow_autolog.py --large
 pip install 'tensorflow>=2.0.0rc2'
 pytest --verbose tests/tensorflow/test_tensorflow2_model_export.py --large
-pip install 'tensorflow==1.12'
+pytest --verbose tests/tensorflow_autolog/test_tensorflow2_autolog.py --large
+pip install 'tensorflow==1.14'
```
Comment (smurching) [inline travis/run-large-python-tests.sh]: I wonder if we could simplify things a bit & skip this step of re-installing TF 1.14 by moving running the tests below this line (`pytest  tests/azureml`, `pytest  tests/models`, etc) to before we install TF 2.0.0

File: travis/run-large-python-tests.sh [hunk 4]
```diff
@@ -34,11 +34,12 @@ pytest --verbose tests/sagemaker/mock --large
 pytest --verbose tests/sklearn --large
 pytest --verbose tests/spark --large
 pytest --verbose tests/tensorflow/test_tensorflow_model_export.py --large
+pytest --verbose tests/tensorflow_autolog/test_tensorflow_autolog.py --large
 pip install 'tensorflow>=2.0.0rc2'
```
Comment (smurching) [inline travis/run-large-python-tests.sh]:
```suggestion
pip install 'tensorflow>=2.0.0'
```

**Review by smurching — COMMENTED** [review]
Just some small comments but otherwise looks good!

File: travis/run-large-python-tests.sh (hunk 3 above)
Comment (juntai-zheng) [reply travis/run-large-python-tests.sh]: Just a heads up, will have to move the keras_autolog test up too, since they don't work with TF 2.0 as of now.

Comment (juntai-zheng) [reply travis/run-large-python-tests.sh]: Gotcha, will do

**Review by smurching — APPROVED** [review]
LGTM thanks @juntai-zheng

**Pull request merged**